### ISSUE TITLE
[CI] RestClusterInfoActionCancellationIT testGetMappingsCancellation failing

### ISSUE THREAD

**Issue by fcofdez** [elastic/elasticsearch#91248]
**Build scan:**
https://gradle-enterprise.elastic.co/s/7oelq7yveddp6/tests/:qa:smoke-test-http:javaRestTest/org.elasticsearch.http.RestClusterInfoActionCancellationIT/testGetMappingsCancellation

**Reproduction line:**
`./gradlew ':qa:smoke-test-http:javaRestTest' --tests "org.elasticsearch.http.RestClusterInfoActionCancellationIT.testGetMappingsCancellation" -Dtests.seed=FA6ACE673DF705E6 -Dtests.locale=he -Dtests.timezone=America/Asuncion -Druntime.java=8`

**Applicable branches:**
7.17

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.http.RestClusterInfoActionCancellationIT&tests.test=testGetMappingsCancellation

**Failure excerpt:**
```
junit.framework.AssertionFailedError: found no cancellable tasks

  at junit.framework.Assert.fail(Assert.java:57)
  at junit.framework.Assert.assertTrue(Assert.java:22)
  at junit.framework.TestCase.assertTrue(TestCase.java:192)
  at org.elasticsearch.test.TaskAssertions.lambda$assertAllCancellableTasksAreCancelled$2(TaskAssertions.java:73)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1141)
  at org.elasticsearch.test.TaskAssertions.assertAllCancellableTasksAreCancelled(TaskAssertions.java:57)
  at org.elasticsearch.http.RestClusterInfoActionCancellationIT.runTest(RestClusterInfoActionCancellationIT.java:83)
  at org.elasticsearch.http.RestClusterInfoActionCancellationIT.testGetMappingsCancellation(RestClusterInfoActionCancellationIT.java:46)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:748)

```

**Comment by pxsalehi** [comment]
Fortunately, there are some trace logs in this failure. Last time for a similar failure that shares some code with this we just increased timeout and log level (https://github.com/elastic/elasticsearch/issues/88884). However, the log messages I've added is inaccurate and when in the log you see:
```
[testGetMappingsCancellation] --> found task with prefix [indices:admin/mappings/get] marked as cancelled: [org.elasticsearch.tasks.CancellableTask@26fb7fa]
```
It doesn't mean we have actually verified that the task is marked as cancellable but rather just that we have seen the task. I will fix that and try to add more logs if this happens again.

What I understand from the logs is that the task cancellation goes through "very quickly", and by the time we get to `assertAllCancellableTasksAreCancelled`, the tasks are unregistered. On `node_t1` the task is registered with id 18, and on `node_t0` as a child of it with id 37. There is cancelation log for task18 on t1, but no such log for task37 on t0. But there are unregistration log messages for both tasks! So my thinking is that first cancelling the task and then relying on `assertAllCancellableTasksAreCancelled` to see the state where there is at least one task on the task manager and for all seen tasks they are cancelled, might be a bit racy. To me it seems possible that that big assertion block could entirely miss seeing those in-between states and just see no tasks for the given action in the task manager instances. 

I was able to reproduce it only once locally by adding some big sleep between the cancellation and `assertAllCancellableTasksAreCancelled` call. But it doesn't reproduce consistently in that way.

If this is the case, we could at lease call the cancellation after starting the assertBusy block (i.e. assertAllCancellableTasksAreCancelled) (somehow in the background)? Although that doesn't fully address the assertion block missing the correct state, I think it will mitigate it a bit.
